### PR TITLE
Hpa improv refactor replica calc test

### PR DESCRIPTION
**Pull request by jbartosik**
**What this PR does / why we need it**: prepareTestClient generates 4 fake clients, using replicaCalcTestCase object. This PR extracts a separate helper for generating each fake independently.

**Which issue(s) this PR fixes**

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Review by DirectXMan12 — CHANGES_REQUESTED** [review]
these should just be methods on the test case object.  Then you don't need double-pointer arguments, among other things.

**Comment by jbartosik** [comment]
Done
/retest

File: pkg/controller/podautoscaler/replica_calculator_test.go [hunk 1]
```diff
@@ -197,7 +202,7 @@ func (tc *replicaCalcTestCase) prepareTestClient(t *testing.T) (*fake.Clientset,
 			return true, nil, fmt.Errorf("no custom metrics specified in test client")
 		}
 
-		assert.Equal(t, tc.metric.name, getForAction.GetMetricName(), "the metric requested should have matched the one specified")
+		assert.Equal(t, (tc.metric).name, getForAction.GetMetricName(), "the metric requested should have matched the one specified")
```
Comment (DirectXMan12) [inline pkg/controller/podautoscaler/replica_calculator_test.go]: what's up with the extra parens?  those aren't necessary

**Review by DirectXMan12 — CHANGES_REQUESTED** [review]
minor nit, otherwise LGTM

**Review by DirectXMan12 — CHANGES_REQUESTED** [review]
another minor nit

File: pkg/controller/podautoscaler/replica_calculator_test.go (hunk 1 above)
Comment (jbartosik) [reply pkg/controller/podautoscaler/replica_calculator_test.go]: Removed

File: pkg/controller/podautoscaler/replica_calculator_test.go [hunk 2]
```diff
@@ -99,7 +98,7 @@ func (tc *replicaCalcTestCase) prepareTestClient(t *testing.T) (*fake.Clientset,
 		}
 		for i := 0; i < podsCount; i++ {
 			podReadiness := v1.ConditionTrue
-			if tc.podReadiness != nil && i < len(tc.podReadiness) {
+			if i < len(tc.podReadiness) {
```
[the remark this answers is not in the document]
Comment (jbartosik) [reply pkg/controller/podautoscaler/replica_calculator_test.go]: Done.

**Comment by DirectXMan12** [comment]
alright, squash this down into one commit (the diff is fairly clean, and it makes sense together), and we're all set and this can merge.

**Pull request merged**